### PR TITLE
Fix matrix decompose (wxyz quaternion)

### PR DESCRIPTION
**Pull request by stephomi**
Since https://github.com/g-truc/glm/pull/1069, the default behavior of `operator[]` has changed, and `decompose` is relying on it.

It also means that `decompose` was previously broken when `GLM_FORCE_QUAT_DATA_WXYZ` was used.

If you don't want preprocessor statement (or avoid `operato[]`), I can change the code to:
```
T qi = static_cast<T>(0.5) * root;
root = static_cast<T>(0.5) / root;
T qj = root * (Row[i][j] + Row[j][i]);
T qk = root * (Row[i][k] + Row[k][i]);

Orientation.w = root * (Row[j][k] - Row[k][j]);

if (i == 0) {
	Orientation.x = qi;
	Orientation.y = qj;
	Orientation.z = qk;
} else if (i == 1) {
	Orientation.y = qi;
	Orientation.z = qj;
	Orientation.x = qk;
} else {
	Orientation.z = qi;
	Orientation.x = qj;
	Orientation.y = qk;
}
```

### CODE REVIEW

**Comment by stephomi** [comment]
Oh it also explains https://github.com/g-truc/glm/pull/1046

**Comment by christophe-lunarg** [comment]
@stephomi, would your fix resolve #1046 too?

**Comment by stephomi** [comment]
I believe so, can't

**Comment by stephomi** [comment]
Yes I think so!

TLDR; previous app (or glm internal codebase) that was relying on `operator[]`or `value_ptr` is probably broken (similarly to `decompose`) since https://github.com/g-truc/glm/pull/1069, since the default is now WXYZ.

Usually we don't operate directly on quaternion but I was typically using `value_ptr` for file serialization, so it broke when I tried to read legacy file. Easy fix though, I'm now using algebric access (`.x`) as it's safer.

I do welcome the recent change to make WXYZ consistent, it's cleaner!

However with `GLM_FORCE_QUAT_DATA_XYZW`, the constructor is still accepting wxyz.
There is a PR about this https://github.com/g-truc/glm/pull/1074, but I don't have a strong opinion on this (like most people I don't use this flag).
It can make sense to just have a flag that change the order in memory while having a consistent wxyz constructor but I don't know if it's the excepted behavior.
